test/Microsoft.AspNetCore.Server.KestrelTests/EngineTests.cs

```diff
@@ -1125,6 +1096,7 @@ public void ConnectionCanReadAndWrite(TestServiceContext testContext)
                     for (int i = 0; i < 100; i++)
                     {
                         await response.WriteAsync(largeString, lifetime.RequestAborted);
+                        registrationWh.Wait(1000);
```

Comment:
Tough one. It means that it would take at least 100 seconds to fail if we broke it, but as flaky as this test is particularly on the CI server, I don't particularly mind.

In practice, this test passes in less than a second on my machine.